colossalai/auto_parallel/tensor_shard/utils/device/get_one_alpha_beta.py

```diff
@@ -0,0 +1,130 @@
+import math
+import socket
+import subprocess
+import time
+
+import torch
+import torch.distributed as dist
+
+MB = int((1 << 10) * 1e3)
+GB = int((1 << 20) * 1e3)
+Byte = 4
+FRAMEWORK = 20 / 1e6
+
+
+def store(time, bandwidth):
+    f = open("tmp.txt", "w")
+    f.write(str(time) + ',' + str(bandwidth))
+    f.close()
+
+
+def load():
+    f = open("tmp.txt", "r")
+    ln = f.readline().split(",")
+    f.close()
+    return (float(ln[0]), float(ln[1]))
+
+
+def execute_cmd(cmd):
+    cmd = ' '.join(cmd)
+    print(cmd)
+    subprocess.call(cmd, shell=True)
+
+
+def test(wsize, nbytes, type):
+    warmup = 5
+    repeat = 25
+    rank = dist.get_rank()
+    torch.cuda.set_device(rank)
+    src_device_num = 0
+    device = torch.device("cuda", rank)
+    hostname = socket.gethostname()
+
+    gpu = f"[{hostname}-{rank}]"
+
+    buf = torch.randn(nbytes // 4).to(device)
+
+    torch.cuda.synchronize()
+    # warmup
+    for _ in range(warmup):
+        if type == "a":
+            dist.all_reduce(buf, op=dist.ReduceOp.SUM)
+        elif type == "b":
+            dist.broadcast(buf, src=src_device_num)
+    torch.cuda.synchronize()
+
+    dist.barrier()
+    begin = time.perf_counter()
+    for _ in range(repeat):
+        if type == "a":
+            dist.all_reduce(buf, op=dist.ReduceOp.SUM)
+        elif type == "b":
+            dist.broadcast(buf, src=src_device_num)
+    torch.cuda.synchronize()
+    end = time.perf_counter()
+    dist.barrier()
+
+    if rank == 0:
+        avg_time_s = (end - begin) / repeat - FRAMEWORK
+        alg_band = nbytes / avg_time_s
+        if type == "b":
+            bus_band = alg_band
+        elif type == "a":
+            bus_band = 2 * (wsize - 1) / wsize * alg_band
+        store(avg_time_s, alg_band)
+        print(
+            f"{gpu}, Bytes: {nbytes} B,Time: {round(avg_time_s * 1e6,2)} us, Bus bandwidth: {round(bus_band / GB,2)} GB/s"
+        )
+        return (avg_time_s, alg_band)
+
+
+def test_latency(wsize, it=3, type="a"):
```

Comment:
Same for all functions starting with `test`.

Reply:
Function names have been modified to profile_xxx